hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java

```diff
@@ -1425,6 +1425,13 @@ protected void update(final AssignmentManager am) {
       this.statTimestamp = EnvironmentEdgeManager.currentTime();
       update(regionStates.getRegionsStateInTransition(), statTimestamp);
       update(regionStates.getRegionFailedOpen(), statTimestamp);
+
+      if (ritsOverThreshold != null && !ritsOverThreshold.isEmpty()) {
+        LOG.trace("RIT hashes and states: {}",
+          ritsOverThreshold.entrySet().stream()
+            .map(e -> e.getKey() + ":" + e.getValue().getState().name())
```

Comment:
I think this can be a debug(), fwiw.

Also, the pretty printing would be evaluated even if TRACE/DEBUG is disabled, so you might want to surround with a check.

if (LOG.isDebugEnabled() && ritsOverThreshold != null && !ritsOverThreshold.isEmpty()) {
  LOG.debug(...
}

Reply:
Oh yes, nice catch. Good to guard with `isDebugEnabled()` 👍